src/DynamoPackages/PackageLoader.cs

```diff
@@ -276,7 +276,7 @@ private void TryLoadPackageIntoLibrary(Package package)
                 Log($"Loaded Package {package.Name} {package.VersionName} from {package.RootDirectory}");
                 try
                 {
-                    if (dynamoVersion.Major == 3 && Version.Parse(package.EngineVersion).Major < 3)
+                    if (dynamoVersion.Major == 4 && Version.Parse(package.EngineVersion).Major < 3)
```

Comment:
not related to the test but i am assuming, we will support packages with engine 3 and 4 for Dynamo 4.x.

Reply:
`dynamoVersion.Major >= 3` should fix this for all future major versions

Reply:
@QilongTang I was suggesting `>=3` as if you see the next condition this will basically throw the warning for all packages published using 2.x, when installed in anything >=3.x.

Reply:
I think this check was majorly due to the switch from net48 to net8, which had a greater impact on packages, moving forward, for packages published using 3.x+ this may or may not be the case, as the upgrades have much less breaking changes for example net8-net10.